src/Utility/Function.h

```diff
@@ -151,6 +151,9 @@ class FunctionBase
   //! \brief Empty destructor.
   virtual ~FunctionBase() {}
 
+  //! \brief Returns the function type flag.
+  virtual unsigned char getType() const = 0;
```

Comment:
i think an enum would be a lot more readable.

Reply:
Maybe, but then I have to think about some good enum value names..